#!/usr/bin/env python
PACKAGE = "yahboomcar_bringup"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("SetLoop", bool_t, 0, "SetLoop", False)
gen.add("Linear", double_t, 0, "Linear in robot", 0.5, 0, 1.0)
gen.add("Angular", double_t, 0, "Angular in robot", 1.0, 0, 5.0)
gen.add("Length", double_t, 0, "Length in limit", 1.0, 0, 2.0)
gen.add("Angle", double_t, 0, "Angle in limit", 360.0, 0, 360.0)
gen.add("LineScaling", double_t, 0, "Line Scaling", 1.1, 0, 2.0)
gen.add("RotationScaling", double_t, 0, "Rotation Scaling", 0.75, 0, 2.0)
gen.add("LineTolerance", double_t, 0, "Line Tolerance", 0.1, 0, 3.0)
gen.add("RotationTolerance", double_t, 0, "Rotation Tolerance", 0.3, 0, 5.0)
gen.add("ResponseDist", double_t, 0, "ResponseDist in limit", 0.6, 0.0, 8.0)
gen.add("LaserAngle", int_t, 0, "LaserAngle in limit", 30, 10, 180)

command_enum = gen.enum([gen.const("LengthTest", int_t, 0, "first data"),
                       gen.const("AngleTest", int_t, 1, "two data"),
                       gen.const("Triangle", int_t, 2, "three datat"),
                       gen.const("Square", int_t, 3, "four data"),
                       gen.const("Circle", int_t, 4, "five data"),
                       gen.const("Parallelogram", int_t, 5, "six data")], "An enum to set size")
gen.add("Command", int_t, 0, "A size parameter which is edited via an enum", 3, 0, 5, edit_method=command_enum)

gen.add("Switch", bool_t, 0, "Switch in robot", False)
exit(gen.generate(PACKAGE, "YahboomCarPatrol", "PatrolParam"))
